### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by blase147**
In this PR, I achieved the following;

- Created a class Nameable.
- Implemented a method called correct_name that will raise a NotImplementedError.
- Turned my Person class to Nameable
- Person class inherited from Nameable
- Method correct_name implemented which returned the name attribute.
- Prepared base Decorator
- Base decorator inherited from Nameable.
- Assigned a nameable object from params to an instance variable.
- Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
- Prepared CapitalizeDecorator and TrimmerDecorator
- Created a class that inherits from the base Decorator class.
- Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
- Created a class that inherits from the base Decorator class.
- Implemented a method correct_name that ensures that the output of @nameable.correct_name has a maximum of 10 characters.

### CODE REVIEW

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @blase147, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Professional Readme
✔️ Correct Giflow

## Required 
- [ ] Good job in trimming the name however, you have to follow all the requirements stated in the project requirements. Make sure you have followed all the requirements.

Screenshot 2023-01-09 at 12 53 28 PM

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by blase147** [comment]
> ## Required Changes ♻️
> Hi @blase147, This is @sja-thedude your code reviewer.
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
> 
> 768070f3-bc9a-4aba-b8bc-25793e00ea1e_text [ 768070f3-bc9a-4aba-b8bc-25793e00ea1e_text ](https://user-images.githubusercontent.com/70269507/159483175-2f61461d-b352-4cb4-98f8-b173e05ab5ff.gif) [ ](https://user-images.githubusercontent.com/70269507/159483175-2f61461d-b352-4cb4-98f8-b173e05ab5ff.gif)
> 
> ## Highlights
> ✔️ All linters passing ✔️ Professional Readme ✔️ Correct Giflow
> 
> ## Required
> * [ ]  Good job in trimming the name however, you have to follow all the requirements stated in the project requirements. Make sure you have followed all the requirements.
> 
> Screenshot 2023-01-09 at 12 53 28 PM
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by blase147** [comment]
Please can you be more precise as to the requirements. Because I feel I followed all requirements

**Comment by blase147** [comment]
Thanks for the feedback. Please @sja-thedude  can you be more precise as to the requirements? Because I feel I followed all requirements

**Comment by sja-thedude** [comment]
> Thanks for the feedback. Please @sja-thedude  can you be more precise as to the requirements? Because I feel I followed all requirements

As per the requirements you should have a class Nameable.
Which should have implement a method correct_name.

Make sure that you have a class person which inherits nameable.
And have correct_name implemented.

Next you should have decorator class which inherits from nameable.

Also you should have class for capitalize decorator and trimmer decorator.

So make sure you follow all the guidelines with this.

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+    puts @nameable.correct_name.capitalize
```
Comment (mckent05) [inline capitalize_decorator.rb]: - According to the project requirements, please return the capitalized name, I suggest that you remove this line of code 🙏 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @blase147, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @blase147, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**